### ISSUE TITLE
[P1][M8] #pragma GCC diagnostic push/pop/ignored/warning/error not implemented

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#110]
## Bug

CCC silently ignores all `#pragma GCC diagnostic` directives. These are ubiquitous in real-world code and system headers to suppress specific warnings around known-safe constructs.

Currently in `src/frontend/preprocessor/pragmas.rs` (line ~63): "Other pragmas (GCC, diagnostic, etc.) are silently ignored."

## Repro

```c
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-variable"
int unused;  // Should NOT warn
#pragma GCC diagnostic pop
int also_unused;  // Should warn (if -Wunused-variable enabled)
```

## Impact

Nearly every non-trivial C project uses `#pragma GCC diagnostic` to manage warnings. System headers (glibc, musl) use them extensively. Without this, compiling real code produces a flood of false warnings or requires disabling warnings entirely.

## Fix

1. In the preprocessor, parse `#pragma GCC diagnostic push|pop|ignored|warning|error "-Wflag-name"`
2. Maintain a diagnostic state stack (push/pop)
3. Pass the current diagnostic state to sema so warning emission checks it
4. Map `-Wflag-name` strings to `WarningKind` variants

## Acceptance Criteria

- [ ] `push` and `pop` create/restore diagnostic state
- [ ] `ignored` suppresses the named warning
- [ ] `warning` re-enables as warning
- [ ] `error` promotes to error
- [ ] Unknown warning names silently ignored (GCC behavior)
- [ ] Integration with existing `WarningKind` system

Part of [MILESTONE] M8: Real-World Code Compatibility (#109)

**Comment by ChaseWNorton** [comment]
Investigating pragma GCC diagnostic implementation. Checking preprocessor pragma handling, diagnostic engine, and WarningKind system.
